tests/integration/test_logcollector/test_location/test_location.py

```diff
@@ -0,0 +1,246 @@
+        {'location': '/tmp/wazuh-testing/non-existent.txt',
+         'files': ['/tmp/wazuh-testing/non-existent.txt'],
+         'log_format': 'syslog', 'file_type': 'non_existent_file'},
+        {'location': '/tmp/wazuh-testing/*',
+         'files': ['/tmp/wazuh-testing/foo.txt', '/tmp/wazuh-testing/bar.log',
+                   '/tmp/wazuh-testing/test.yaml', '/tmp/wazuh-testing/ñ.txt',
+                   '/tmp/wazuh-testing/テスト.txt', '/tmp/wazuh-testing/ИСПЫТАНИЕ.txt',
+                   '/tmp/wazuh-testing/测试.txt', '/tmp/wazuh-testing/اختبار.txt'],
+         'log_format': 'syslog', 'file_type': 'wildcard_file'},
+        {'location': '/tmp/wazuh-testing/Testing white spaces',
+         'files': ['/tmp/wazuh-testing/Testing white spaces'], 'log_format': 'syslog',
+         'file_type': 'single_file'},
+        {'location': '/tmp/wazuh-testing/test.*',
+         'files': ['/tmp/wazuh-testing/test.txt', '/tmp/wazuh-testing/test.log'],
+         'log_format': 'syslog', 'file_type': 'wildcard_file'},
+        {'location': '/tmp/wazuh-testing/c*test.txt',
+         'files': ['/tmp/wazuh-testing/c1test.txt', '/tmp/wazuh-testing/c2test.txt',
+                   '/tmp/wazuh-testing/c3test.txt'], 'log_format': 'syslog',
+         'file_type': 'wildcard_file'},
+        {'location': '/tmp/wazuh-testing/duplicated/duplicated.txt',
+         'files': ['/tmp/wazuh-testing/duplicated/duplicated.txt'],
+         'log_format': 'syslog', 'path_2': '/tmp/wazuh-testing/duplicated/duplicated.txt',
+         'file_type': 'duplicated_file'},
+        {'location': '/tmp/wazuh-testing/file.log-%Y-%m-%d',
+         'files': [f'/tmp/wazuh-testing/file.log-{date}'], 'log_format': 'syslog',
+         'file_type': 'single_file'},
+        {'location': '/tmp/wazuh-testing/multiple-logs/*', 'files': ['/tmp/wazuh-testing/multiple-logs/multiple'],
+         'log_format': 'syslog', 'file_type': 'multiple_logs'}
+    ]
+
+
+# Configuration data
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f"{x['LOCATION'], x['LOG_FORMAT']}" for x in parameters]
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+
+# Fixtures
+@pytest.fixture(scope="module")
+def create_directory():
+    """Create expected directories."""
+    if sys.platform == 'win32':
+        os.makedirs(fr'{temp_dir}\wazuh-testing\multiple-logs', exist_ok=True)
+        os.makedirs(fr'{temp_dir}\wazuh-testing\depth1\depth2', exist_ok=True)
+        os.makedirs(fr'{temp_dir}\wazuh-testing\duplicated', exist_ok=True)
+    else:
+        os.makedirs('/tmp/wazuh-testing/multiple-logs', exist_ok=True)
+        os.makedirs('/tmp/wazuh-testing/depth1/depth2', exist_ok=True)
+        os.makedirs('/tmp/wazuh-testing/duplicated', exist_ok=True)
+    yield
+
+    if sys.platform == 'win32':
+        rmtree(fr'{temp_dir}\wazuh-testing', ignore_errors = True)
+    else:
+        rmtree('/tmp/wazuh-testing')

```

Comment:
```suggestion
    os.makedirs(os.path.join(temp_dir, 'wazuh-testing', 'multiple-logs'), exist_ok=True)
    os.makedirs(os.path.join(temp_dir, 'wazuh-testing', 'depth1', 'depth2'), exist_ok=True)
    os.makedirs(os.path.join(temp_dir, 'wazuh-testing', 'duplicated'), exist_ok=True)
    yield

    rmtree(os.path.join(temp_dir, 'wazuh-testing') , ignore_errors = True)
```

Using `gettempdir` at line 27, you can simplify this block by using `os.path.join` instead of an `if/else` block. You can apply this change in all the new tests, including the dictionaries with the test cases.

Reply:
Fixed in https://github.com/wazuh/wazuh-qa/commit/cd5a1cb69fb6ecb73971f273482cbf0f134a0960